fmn/config.py

```diff
@@ -1,6 +1,341 @@
+                'task_serializer': 'json',
+                'task_default_queue': 'fmn.tasks.unprocessed_messages',
+                'beat_schedule': {
+                    'process-digests': {
+                        'task': 'fmn.tasks.batch_messages',
+                        'schedule': 60.0,
+                    },
+                    'process-confirmations': {
+                      'task': 'fmn.tasks.confirmations',
+                      'schedule': 15.0,
+                    },
+                },
+            },
+            'validator': None,
+        },
+        'logging': {
+            'value': {
+                'version': 1,
+                'formatters': {
+                    'bare': {
+                        'datefmt': '%Y-%m-%d %H:%M:%S',
+                        'format': '[%(asctime)s][%(name)10s %(levelname)7s] %(message)s'
+                    },
+                },
+                'handlers': {
+                    'console': {
+                        'class': 'logging.StreamHandler',
+                        'formatter': 'bare',
+                        'stream': 'ext://sys.stdout',
+                    },
+                },
+                'loggers': {
+                    'fmn': {
+                        'level': 'INFO',
+                        'propagate': False,
+                        'handlers': ['console'],
+                    }
+                },
+                'root': {
+                    'level': 'WARNING',
+                    'handlers': ['console'],
+                },
+            },
+            'validator': None,
+        },
+    }
+
+    def __getitem__(self, *args, **kw):
+        if not self.loaded:
+            self.load_config()
+        return super(FmnConfig, self).__getitem__(*args, **kw)
+
+    def get(self, *args, **kw):
+        if not self.loaded:
+            self.load_config()
+        return super(FmnConfig, self).get(*args, **kw)
+
+    def pop(self, *args, **kw):
+        if not self.loaded:
+            self.load_config()
+        return super(FmnConfig, self).pop(*args, **kw)
+
+    def copy(self, *args, **kw):
+        if not self.loaded:
+            self.load_config()
+        return super(FmnConfig, self).copy(*args, **kw)
+
+    def load_config(self, settings=None):
+        """
+        Load the configuration either from the config file, or from the given settings.
+
+        args:
+            settings (dict): If given, the settings are pulled from this dictionary. Otherwise, the
+                config file is used.
+        """
+        self._load_defaults()
+        if settings:
+            self.update(settings)
+        else:
+            self.update(fedmsg.config.load_config())
+        self.loaded = True
+        self._validate()
+
+    def _load_defaults(self):
+        """Iterate over self._defaults and set all default values on self."""
+        for k, v in self._defaults.items():
+            self[k] = v['value']
```

Comment:
I'd kinda wished I'd named this key "default" instead of "value". Maybe I'll change that in Bodhi one day. Just a thought.

Reply:
I shall learn from your regrets. I'll make it better, stronger. I have the technology.